dbcon/joblist/tuplehashjoin.cpp

```diff
@@ -116,11 +116,13 @@ TupleHashJoinStep::TupleHashJoinStep(const JobInfo& jobInfo) :
     else
         allowDJS = false;
 
+    numCores = sysconf(_SC_NPROCESSORS_ONLN);
+    if (numCores <= 0)
+        numCores = 8;
```

Comment:
You could have also got this information from resourceManager which does exactly the same thing but also appears to support CGroups

Reply:
will make that change.